teps/0098-workflows.md

```diff
@@ -256,11 +306,164 @@ Why should this TEP _not_ be implemented?
+as the reference implementation evolves.
+
+Cons:
+- It will be difficult to come up with a Workflows conformance spec without first creating a reference implementation
+that supports several SCMs. It's unclear whether designing a good conformance spec is easier than designing a good
+API extensibility mechanism (and later on declaring a conformance spec).
+- Platform builders who want to change only part of the implementation will be forced to recreate the entire implementation.
+  - For example, a platform builder might want to add some custom steps to the process of connecting to a repo, or add their own backing
+  for secrets storage, without rewriting much of the project.
+
+### Adopt Pipelines as Code as Workflows Replacement
+
+In this solution, originally proposed in [#866](https://github.com/tektoncd/community/issues/883),
+Pipelines as Code would be transferred to the tektoncd organization,
+built and released as a Tekton project, and likely renamed to "Workflows", since the scope
+of the Workflows project is larger than storing Pipelines in version control.
+The Tekton community would maintain the project and shape its future to meet the goals of the Workflows project.
+The doc [Pipelines as Code -> Tekton Workflows](https://docs.google.com/document/d/1JAnXFkvs4aeciAJEsl7454e0J3ZHbL1yCPncMBHlTtM/)
+compares Pipelines as Code to the goals of the Workflows project.
+
+Suggested steps:
+1. Design and implement a solution for parameter substitution in Pipelines as Code that's Tekton conformant.
+(The current [syntax](https://pipelinesascode.com/docs/guide/authoringprs/#authoring-pipelineruns-in-tekton-directory)
+for variable expansion isn't Tekton conformant.)
+1. Move the project to the tektoncd org, rename it, run tests on Tekton infrastructure, and publish releases to the tekton-releases GCP project.
+Use tekton standards for code, conduct, and contributions. Choose and document a Tekton stability level for the existing API.
+1. Update the project to make more extensive use of remote resolution. For example, we may want to keep the ability to refer to Tekton CRDs in the same
+repo using local paths, but encourage the use of the built-in hub resolvers rather than using the Pipelines as Code
+[hub resolution](https://pipelinesascode.com/docs/guide/resolver/#tekton-hubhttpshubtektondev) and the built-in git resolver rather than using
+the Pipelines as Code [remote URL syntax](https://pipelinesascode.com/docs/guide/resolver/#remote-http-url).
+(Note: this work is already planned for Pipelines as Code.)
+1. Create a Workflow CRD that allows users to configure much of the behavior that's currently configured through annotations, such as which events
+trigger a PipelineRun and concurrency controls to apply. We may choose to rework the existing implementation so that configuring a PipelineRun
+with annotations would be implemented by creating a Workflow under the hood. We'd support the existing API (PipelineRun with annotations) for at
```

Comment:
Having the annotations part of the PipelineRun bring another advantage, where a user can easily drop a PipelineRun from one project to another project, without having to care about which part to take in the Workflow CRD.

It makes it pretty simple to understand and to write as well..

But annotations would not scale to at least the advanced [event matching feature](https://pipelinesascode.com/docs/guide/authoringprs/#advanced-event-matching),  I have just created a [issue](https://github.com/openshift-pipelines/pipelines-as-code/issues/1049) proposing a solution for it.

Reply:
Can you elaborate on how annotations make it easier to move a PipelineRun from project to project? Is the idea that if we had a workflow api that looked something like this:
```
events:
- source:
      repo: pipelines
   type: pull_request
pipelineRun:
....
```
that you would have to rewrite the bits of the spec related to which repo to use?

Reply:
I do agree that annotations make it easy to experiment with but I think for core CI features like filtering etc, having it in their own fields has many advantages e.g. better validation, support for more complex syntax, less verbosity etc.